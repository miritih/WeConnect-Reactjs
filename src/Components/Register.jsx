import React from 'react';
// import $ from 'jquery'
// import components 
import RegisterForm from './forms/Register';
import NavBar from './NavBar';

class Register extends React.Component {
  render() {
    return (
      <div>
      <NavBar 
        history={this.props.history} 
        location={this.props.location} 
      />
      <RegisterForm history={this.props.history} />
    </div>
    );
  }
}

export default Register;
