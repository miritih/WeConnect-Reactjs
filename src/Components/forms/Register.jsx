import React from 'react';
import axios from 'axios';
import Authservice from '../Auth/AuthService';
import { toast } from 'react-toastify';
import { Link } from 'react-router-dom';
import { baseURL } from '../../utils/Config';

class RegisterForm extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      username: '',
      email: '',
      password: '',
      cpassword: '',
      first_name: '',
      last_name: '',
      success: [],
      errors: {}
    };
    this.auth = new Authservice();
    this.handleChange = this.handleChange.bind(this);
    this.confirmPass = this.confirmPass.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleChange(e) {
    this.setState({
      [e.target.name]: e.target.value,
    });
  }
  confirmPass(e) {
    this.setState({
      cpassword: e.target.value,
    });
  }
  handleSubmit(event) {
    event.preventDefault();
    const user = {
      username: this.state.username,
      email: this.state.email,
      password: this.state.password,
      first_name: this.state.first_name,
      last_name: this.state.last_name
    };

    axios({
      method: 'post',
      url: 'auth/register',
      data: user,
      baseURL: baseURL,
      responseType: 'json',
      headers: {
        'Content-Type': 'application/json'
      },
    }).then((response) => {
      if (response.status >= 200 && response.status < 300) {
        this.props.history.replace('/login');
        toast.success(({ closeToast }) => <div>
          <h3>Success</h3>
              <p>Your Account was successfully Created<br/> 
              Login now with your username and password
              </p>
          </div>);
      }
    }).catch((error) => {
      if (error.response !== undefined) {
        this.setState({ errors: error.response.data['Errors'] });
      }
      else {
        toast.error(({ closeToast }) => <div>
          <h3>Opps!!</h3>
              <p>Sorry! Something went wrong. If the problem persist, contact support</p>
          </div>);
      }
    });
  }
  componentDidMount() {
    if (this.auth.isLoggedIn())
      this.props.history.replace('/');
  }
  render() {
    return (
      <div className="container wrapper">
        <div className="main-center">
              <div className="card">
                <div className="card-body">
                <h3>Create Account</h3>
                   <form className="form" noValidate onSubmit={this.handleSubmit}>
                      <label htmlFor="username" >Username</label>
                      <input 
                      type="text" 
                      className={"form-control " + (
                      this.state.errors['username'] === undefined ? "" : "is-invalid"
                      )}
                      name="username"
                      value= {this.state.username}
                      onChange={this.handleChange}
                      />
                       {(this.state.errors['username'] !== undefined) && 
                       (<small className="text-danger">
                       {this.state.errors['username'].join()}
                       <br/>
                       </small> )
                       }             
                      <label htmlFor="email">Email</label>
                      <input
                      type="email"
                      className={"form-control " + (
                      this.state.errors['email'] === undefined ? "" : "is-invalid"
                      )}
                      name="email"
                      value= {this.state.email}
                      onChange={this.handleChange}
                      />
                       {(this.state.errors['email'] !== undefined) && 
                       (<small className="text-danger">
                       {this.state.errors['email'].join()}
                       <br/>
                       </small> )
                       }  
                      <label htmlFor="first_name">First Name</label>
                      <input 
                      type="text" 
                      className = {
                        "form-control " + (
                          this.state.errors['first_name'] === undefined ? "" : "is-invalid"
                        )
                      }
                      name="first_name"
                      value= {this.state.first_name}
                      onChange={this.handleChange}
                      />
                       {(this.state.errors['first_name'] !== undefined) && 
                       (<small className="text-danger">
                       {this.state.errors['first_name'].join()}
                       <br/>
                       </small> )
                       }  
                      <label htmlFor="last_name">Last Name</label>
                      <input
                      type="text"
                      className={"form-control " + (
                      this.state.errors['last_name'] === undefined ? "" : "is-invalid"
                      )}
                      name="last_name"
                      value= {this.state.lastModifiedDate}
                      onChange={this.handleChange}  
                      />
                       {(this.state.errors['last_name'] !== undefined) && 
                       (<small className="text-danger">
                       {this.state.errors['last_name'].join()}
                       <br/>
                       </small> )
                       } 
                      <label htmlFor="password">Password</label>
                      <input 
                      type="password" 
                      ref="password"
                      className={"form-control " + (
                      this.state.errors['password'] === undefined ? "" : "is-invalid"
                      )}
                      name="password"
                      onChange={this.handleChange}
                      />
                       {(this.state.errors['password'] !== undefined) && 
                       (<small className="text-danger">
                       {this.state.errors['password'].join()}
                       <br/>
                       </small> )
                       }
                      <label htmlFor="confirm_password">Confirm Password</label>
                      <input 
                      type="password" 
                      className={"form-control " + 
                      (this.state.cpassword !== this.state.password ? "is-invalid" : ""
                      )}
                      name="confirm_password"
                      onChange={this.confirmPass}
                      />
                      {(this.state.cpassword !== this.state.password) && 
                       (<small className="text-danger">
                        Password mismatch
                       <br/>
                       </small> )
                       }
                      <br/>
                      <button
                        type="submit"
                        className="btn btn-lg btn-success btn-block btn-signin">Register
                      </button>
                      <p className="message">Alredy registered? <Link to="/login">Signin</Link></p>
                   </form>
                </div>
              </div>
          </div>
      </div>
    );
  }
}

export default RegisterForm;
