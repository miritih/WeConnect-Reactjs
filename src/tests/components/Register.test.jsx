import React from 'react';
import ReactDOM from 'react-dom';
import Register from '../../Components/Register';
import { MemoryRouter as Router } from 'react-router-dom';
// import renderer for snapshot testing
import renderer from 'react-test-renderer';

describe('Register', () => {
  it('renders without crashing', () => {
    const pathname = {
      pathname: "/"
    };
    const rendered = renderer.create(
      <Router><Register location={pathname} /></Router>
    );
    expect(rendered.toJSON()).toMatchSnapshot();
  });
});
