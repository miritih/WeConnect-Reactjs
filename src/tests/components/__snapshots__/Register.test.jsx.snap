// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register renders without crashing 1`] = `
<div>
  <div>
    <nav
      className="navbar navbar-expand-lg navhome navbar-light bg-transparent fixed-top"
    >
      <div
        className="container"
      >
        <a
          className="navbar-brand"
          href="/"
          onClick={[Function]}
        >
          <span
            className="white"
          >
            WeConnect
          </span>
        </a>
        <button
          aria-controls="topnav"
          aria-expanded="false"
          aria-label="Toggle navigation"
          className="navbar-toggler"
          data-target="#topnav"
          data-toggle="collapse"
          type="button"
        >
          <span
            className="navbar-toggler-icon"
          />
        </button>
        <div
          className="collapse navbar-collapse"
          id="topnav"
        >
          <ul
            className="navbar-nav ml-auto white"
          >
            <li
              className="nav-item"
            >
              <a
                className="nav-link"
                href="/"
                onClick={[Function]}
              >
                Home
              </a>
            </li>
            <li
              className="nav-item"
            >
              <a
                className="nav-link"
                href="/register"
                onClick={[Function]}
              >
                Signup
              </a>
            </li>
            <li
              className="nav-item"
            >
              <a
                className="nav-link"
                href="/login"
                onClick={[Function]}
              >
                Login
              </a>
            </li>
          </ul>
        </div>
      </div>
    </nav>
  </div>
  <div
    className="container wrapper"
  >
    <div
      className="main-center"
    >
      <div
        className="card"
      >
        <div
          className="card-body"
        >
          <h3>
            Create Account
          </h3>
          <form
            className="form"
            noValidate={true}
            onSubmit={[Function]}
          >
            <label
              htmlFor="username"
            >
              Username
            </label>
            <input
              className="form-control "
              name="username"
              onChange={[Function]}
              type="text"
              value=""
            />
            <label
              htmlFor="email"
            >
              Email
            </label>
            <input
              className="form-control "
              name="email"
              onChange={[Function]}
              type="email"
              value=""
            />
            <label
              htmlFor="first_name"
            >
              First Name
            </label>
            <input
              className="form-control "
              name="first_name"
              onChange={[Function]}
              type="text"
              value=""
            />
            <label
              htmlFor="last_name"
            >
              Last Name
            </label>
            <input
              className="form-control "
              name="last_name"
              onChange={[Function]}
              type="text"
              value={undefined}
            />
            <label
              htmlFor="password"
            >
              Password
            </label>
            <input
              className="form-control "
              name="password"
              onChange={[Function]}
              type="password"
            />
            <label
              htmlFor="confirm_password"
            >
              Confirm Password
            </label>
            <input
              className="form-control "
              name="confirm_password"
              onChange={[Function]}
              type="password"
            />
            <br />
            <button
              className="btn btn-lg btn-success btn-block btn-signin"
              type="submit"
            >
              Register
            </button>
            <p
              className="message"
            >
              Alredy registered? 
              <a
                href="/login"
                onClick={[Function]}
              >
                Signin
              </a>
            </p>
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;
